java/ql/src/semmle/code/java/frameworks/guava/StringUtils.qll

```diff
@@ -0,0 +1,168 @@
+/** Definitions of flow steps through the various string utility fenctions in the Guava framework. */
+
+import java
+private import semmle.code.java.dataflow.FlowSteps
+
+/**
+ * The class `com.google.common.base.Strings`.
+ */
+class TypeGuavaStrings extends Class {
+  TypeGuavaStrings() { this.hasQualifiedName("com.google.common.base", "Strings") }
+}
+
+/**
+ * The class `com.google.common.base.Joiner`.
+ */
+class TypeGuavaJoiner extends Class {
+  TypeGuavaJoiner() { this.hasQualifiedName("com.google.common.base", "Joiner") }
+}
+
+/**
+ * The nested class `Joiner.MapJoiner`.
+ */
+class TypeGuavaMapJoiner extends NestedClass {
+  TypeGuavaMapJoiner() {
+    this.getEnclosingType() instanceof TypeGuavaJoiner and
+    this.hasName("MapJoiner")
+  }
+}
+
+/**
+ * The class `com.google.common.base.Splitter`.
+ */
+class TypeGuavaSplitter extends Class {
+  TypeGuavaSplitter() { this.hasQualifiedName("com.google.common.base", "Splitter") }
+}
+
+/**
+ * The nested class `Splitter.MapSplitter`.
+ */
+class TypeGuavaMapSplitter extends NestedClass {
+  TypeGuavaMapSplitter() {
+    this.getEnclosingType() instanceof TypeGuavaSplitter and
+    this.hasName("MapSplitter")
+  }
+}
+
+/**
+ * A taint preserving method on `com.google.common.base.Strings`.
+ */
+private class GuavaStringsTaintPreservingMethod extends TaintPreservingCallable {
+  GuavaStringsTaintPreservingMethod() {
+    this.getDeclaringType() instanceof TypeGuavaStrings and
+    // static String emptyToNull(String string)
+    // static String emptyToNull(String string)
+    // static String padEnd(String string, int minLength, char padChar)
+    // static String padStart(String string, int minLength, char padChar)
```

Comment:
```suggestion
    // static String padStart(String string, int minLength, char padChar)
    // static String padEnd(String string, int minLength, char padChar)
```
Might be good to either switch them here or in the string set below so both are in the same order.